### ISSUE TITLE
How to prevent repeatedly correcting L1C data when calling `sen2r()` in a while loop?

### ISSUE THREAD

**Issue by philiporlando** [ranghetti/sen2r#214]
I need to create an aggregated raster layer that has fewer than 5% NA values. However, my study area is within the tropics and is very cloudy. As a result, many cells are being masked as NA. I've been trying to use a while loop which iteratively increases the `max_cloud_safe_value` and `timewindow` until enough `SAFE` data has been downloaded for me to aggregate into a single layer (by finding the median values for each band), but the resulting composite rasters still have too many NA values (usually more than 20%). 

I've been using `step_atmcorr = 'l2a'` because `senr()` appears to re-run the atmospheric correction for `L1C` data every loop when setting `step_atmcorr = 'auto'`. Is there an easy fix to avoid reprocessing L1C data into L2A each loop? 

Do you have any other recommendations for producing composite rasters with very low NA percentages within cloudy areas? Maybe I should explore using the `timeperiod = 'seasonal'` parameter?

Thanks again for creating such an awesome open toolbox!

```
while(pct_na > 5) {

    previous_sen_files <- list.files("./data/sentinel-2/sen2r-out/BOA", "\\.tif$", full.names = T)

    sen2r(gui = FALSE,
            timewindow = c(as.Date("2018-06-01), as.Date("2018-07-01")),
            extent = "./data/traverse/study_extent/study_extent.kml",
            extent_name = "myExtent",
            list_prods = c("BOA"),
            max_cloud_safe = i,
            mask_type = "scl_0_1_2_3_7_8_9_10_11", 
            max_mask = 100, 
            mask_smooth = 50,
            mask_buffer = 50,
            res_s2 = "10m",
            extent_as_mask = T,
            parallel = detectCores()-1,
            processing_order = "3", 
            step_atmcorr = "auto", # re-runs atmcorr on l1c each loop? very slow!
            overwrite = F,
            overwrite_safe = F,
            rm_safe = "no",
            path_l1c = "./data/sentinel-2/sen2r-in", path_l2a = "./data/sentinel-2/sen2r-in",
            path_out = "./data/sentinel-2/sen2r-out")

    # Determine NA percentage after downloading more data
    sen_files <- list.files("./data/sentinel-2/sen2r-out/BOA", "\\.tif$", full.names = T)
    if(length(sen_files) > length(previous_sen_files)) {
        s <- stack(sen_files)
        pct_na <- freq_na(s) # I can provide this function if needed
      }   
}
```

**Comment by ranghetti** [comment]
It is strange that, every time, a new L2A is generated by Sen2cCor. I suspect this could be due to the behaviour of Sen2Cor 2.8.0 to change the product name (using the date of processing instead of the ingestion time). 
In the meanwhile I will investigate the issue, try downloading the development branch of the package and launch the GUI to check the dependencies:
```r
remotes::install_github("ranghetti/sen2r", ref = "devel")
library(sen2r)
check_sen2r_deps()
```
Checking for Sen2Cor, you will find the possibility to install the version 2.5.5.
Try using that one and let me know if the issue will still be present.

**Comment by ranghetti** [comment]
Now the issue should be solved both using Sen2Cor 2.5.5 (which is recommended) and 2.8.0.
You can reinstall normally the package and try.
Let me know,

**Comment by philiporlando** [comment]
I reinstalled `sen2r` and installed `sen2cor 2.5.5` and things are working as intended now. Thanks for the help!
